lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceTest.java

```diff
@@ -1288,4 +1283,405 @@ public void disconnect_leave() {
+			channel2.presence.enter("Hello, #1", null);
+			new Helpers.PresenceWaiter(channel2).waitFor(ably2.options.clientId, Action.enter);
+
+			ArrayList<PresenceMessage> receivedMessageStack = new ArrayList<>();
+			channel2.presence.subscribe(new Presence.PresenceListener() {
+				List<PresenceMessage> messageStack;
+
+				@Override
+				public void onPresenceMessage(PresenceMessage[] messages) {
+					messageStack.addAll(Arrays.asList(messages));
+				}
+
+				public Presence.PresenceListener setMessageStack(List<PresenceMessage> messageStack) {
+					this.messageStack = messageStack;
+					return this;
+				}
+			}.setMessageStack(receivedMessageStack));
+
+			/* Start emitting channel with ably client 1 (emitter) */
+			channel1.presence.enter("Hello, #2!", null);
+			channel1.presence.update("Lorem ipsum", null);
+			channel1.presence.update("Dolor sit!", null);
+			channel1.presence.leave(null);
+
+			/* Wait until receiver client (ably2) observes {@code Action.leave}
+			 * is emitted from emitter client (ably1)
+			 */
+			Helpers.PresenceWaiter leavePresenceWaiter = new Helpers.PresenceWaiter(channel2);
+			leavePresenceWaiter.waitFor(ably1.options.clientId, Action.leave);
+
+			/* Validate that,
+			 *	- we received all actions
+			 */
+			assertThat(receivedMessageStack.size(), is(equalTo(3)));
+			for (PresenceMessage message : receivedMessageStack) {
+				assertThat(message.action, isOneOf(Action.update, Action.leave));
+			}
+		} finally {
+			if (ably1 != null) ably1.close();
+			if (ably2 != null) ably2.close();
+		}
+	}
+
+	/**
+	 * <p>
+	 * Validates a client can observe presence messages of other client,
+	 * when they entered to the same channel and observing client subscribed
+	 * to multiple actions.
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
+	public void realtime_presence_subscribe_multiple() throws AblyException {
+		/* Ably instance that will emit presence events */
+		AblyRealtime ably1 = null;
+		/* Ably instance that will receive presence events */
+		AblyRealtime ably2 = null;
+
+		String channelName = "test.presence.subscribe.multiple" + System.currentTimeMillis();
+		EnumSet<PresenceMessage.Action> actions = EnumSet.of(Action.update, Action.leave);
+
+		try {
+			ClientOptions option1 = testVars.createOptions(testVars.keys[0].keyStr);
+			option1.clientId = "emitter client";
+			ClientOptions option2 = testVars.createOptions(testVars.keys[0].keyStr);
+			option2.clientId = "receiver client";
+
+			ably1 = new AblyRealtime(option1);
+			ably2 = new AblyRealtime(option2);
+
+			Channel channel1 = ably1.channels.get(channelName);
+			channel1.attach();
+			(new ChannelWaiter(channel1)).waitFor(ChannelState.attached);
+
+			Channel channel2 = ably2.channels.get(channelName);
+			channel2.attach();
+			(new ChannelWaiter(channel2)).waitFor(ChannelState.attached);
+
+			/* Start listening channel for events with ably client 2 (receiver) */
+			channel2.presence.enter("Hello, #1", null);
```

Comment:
You don't have to be present to subscribe to presence events, I would recommend removing this enter of presence for channel2 as it's unrelated to the test and we should be sure that presence events are published to clients who are not present